### PR TITLE
journal: various fixes to journal_file_read_object()

### PR DESCRIPTION
**Pull request by poettering**
This fixes a bunch of issues:

pread() returns ssize_t, and returns errors in 'errno', handle that
correctly.

More importantly: it might incompletely read data in case we hit
EOF. Check for that, and handle it.

Finally, rename the function to journal_file_read_object_header(), since
it really doesn't read full objects, but only their headers.

Follow-up for: 117e21121e857b4b7d81949542e8dd257265970a

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Ouch, good catch, LGTM

File: src/libsystemd/sd-journal/journal-file.c [hunk 1]
```diff
@@ -126,6 +129,43 @@ int journal_file_tail_end(JournalFile *f, uint64_t *ret_offset) {
...
 
+int journal_file_tail_end_by_mmap(JournalFile *f, uint64_t *ret_offset) {
+        uint64_t p;
+        int r;
+
+        assert(f);
+        assert(f->header);
+        assert(ret_offset);
+
+        /* Same as journal_file_tail_end_by_prad() above, but operates with the usual mmap logic */
```
Comment (daandemeyer) [inline src/libsystemd/sd-journal/journal-file.c]: typo here

**Comment by poettering** [comment]
I added another commit on top. PTAL.

**Comment by DaanDeMeyer** [comment]
I considered the same issue but since I didn't really notice any major regression by using pread(), I left it as is. I did notice major regressions yesterday when I tried to use journal_file_read_object() in some other part of journald so it makes sense to avoid all the syscalls here as well.

**Pull request merged**